src/qt3utils/datagenerators/spectrometers/spectrometer.py

```diff
@@ -0,0 +1,230 @@
+        """
+        return np.nan
+
+    @abc.abstractmethod
+    @sensor_temperature_set_point.setter
+    def sensor_temperature_set_point(self, deg_celsius: float) -> None:
+        """
+        Sets the sensor target temperature in Celsius.
+        """
+        pass
+
+    @abc.abstractmethod
+    @property
+    def exposure_time(self) -> float:
+        """
+        Returns the single frame exposure time (in ms).
+        """
+        return np.nan
+
+    @abc.abstractmethod
+    @exposure_time.setter
+    def exposure_time(self, ms: float) -> None:
+        """
+        Sets the single frame exposure time in milliseconds.
+        """
+        pass
+
+    @property
+    def clock_rate(self) -> float:
+        """
+        The clock rate of a single exposure (1/exposure_time in Hz).
+        """
+        try:
+            _t = self.exposure_time / 1000.0  # Converting from milliseconds to seconds.
+        except Exception as e:
+            self.logger.error(e)
+            _t = 2
+        return 1.0 / _t
+
+    def acquire(
+            self,
+            acquisition_mode: Literal['single', 'step-and-glue', 'kinetic series', 'accumulation'],
+            **kwargs
+    ) -> Tuple[np.ndarray, np.ndarray]:
+        """
+        General method that acquires data depending on the acquisition mode.
+        If you want to add a new mode, overwrite this method, call it via super().acquire(...) and then
+        use an if statement for each different mode you implemented.
+
+        Parameters
+        ----------
+        acquisition_mode : Literal['single', 'step-and-glue', 'kinetic series', 'accumulation']
+            The acquisition mode.
+        **kwargs
+            Additional keyword arguments.
+        Returns
+        -------
+        Tuple[np.ndarray, np.ndarray]
+            The single or multiple spectra and the wavelength array.
+        """
+        if acquisition_mode not in self.ACQUISITION_MODES:
+            raise ValueError(f'Unsupported acquisition mode: {acquisition_mode}')
+        if acquisition_mode == 'single':
+            valid_kwargs = filter_only_valid_kwargs_for_method(self.single_acquisition, kwargs)
+            return self.single_acquisition(**valid_kwargs)
+        if acquisition_mode == 'step-and-glue':
+            valid_kwargs = filter_only_valid_kwargs_for_method(self.step_and_glue_acquisition, kwargs)
+            return self.step_and_glue_acquisition(**valid_kwargs)
+        if acquisition_mode == 'kinetic series':
+            valid_kwargs = filter_only_valid_kwargs_for_method(self.kinetic_series_acquisition, kwargs)
+            return self.kinetic_series_acquisition(**valid_kwargs)
+        if acquisition_mode == 'accumulation':
+            valid_kwargs = filter_only_valid_kwargs_for_method(self.accumulation_acquisition, kwargs)
+            return self.accumulation_acquisition(**valid_kwargs)
+
+    def single_acquisition(self, **kwargs) -> Tuple[np.ndarray, np.ndarray]:
```

Comment:
What wavelengths are you acquiring? Would you just be returning acquire() and get_wavelengths() similar to my acquire_frame method? As you know that will just return the calibrations for wavelengths, in Lightfield this may or may not include the center wavelength as well (will need to double check)

Reply:
@vasilisniaouris Please check the Lightfield documentation in the chm file and search up 'SystemColumnCalibration'. I want to be sure that we actually want to return both the acquired data and the wavelengths since that is what we will do according to the implementation in the princeton datagenerator